src/averageAges.js

```diff
@@ -55,10 +79,46 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const male = 'm';
+  const female = 'f';
+  const men = filterPeopleBySex(people, male);
+  const women = filterPeopleBySex(people, female);
+
+  const mothers = women.filter(woman => (
+    people.some(person => person.mother === woman.name)
+  ));
+
+  const peopleWithMothers = (
+    onlyWithSon
+      ? men
+      : people
+  ).map(person => ({
+    ...person,
+    motherObj: mothers.find(mother => (
+      mother.name === person.mother || null
```

Comment:
The || null operation seems unnecessary because if mother.name does not equal person.mother, undefined will be returned by default.